common/flatpak-run.c

```diff
@@ -2013,14 +1978,15 @@ flatpak_run_add_x11_args (GPtrArray *argv_array,
       *envp_p = g_environ_setenv (*envp_p, "DISPLAY", ":99.0", TRUE);
 
 #ifdef ENABLE_XAUTH
-      int fd;
-      fd = g_file_open_tmp ("flatpak-xauth-XXXXXX", &tmp_path, NULL);
-      if (fd >= 0)
+      g_auto(GLnxTmpfile) xauth_tmpf  = { 0, };
+
+      if (glnx_open_anonymous_tmpfile (O_RDWR | O_CLOEXEC, &xauth_tmpf, NULL))
         {
-          FILE *output = fdopen (fd, "wb");
+          FILE *output = fdopen (xauth_tmpf.fd, "wb");
           if (output != NULL)
             {
-              int tmp_fd = dup (fd);
+	      /* fd is now owned by output, steal it from the tmpfile */
+              int tmp_fd = dup (glnx_steal_fd (&xauth_tmpf.fd));
```

Comment:
This is weird/awkward; it'd be better to do the `FILE*` creation inside `write_xauth()` I think.  But it's not a new issue; we can fix that up after.